realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -182,7 +182,7 @@ public void removeChangeListener(RealmChangeListener listener) {
      * </pre>
      *
      * @return RxJava Observable that only calls {@code onNext}. It will never call {@code onComplete} or {@code OnError}.
-     * @throws UnsupportedOperationException if the required RxJava framework is not on the classpath.
+     * @throws NoClassDefFoundError if the required RxJava framework is not on the classpath.
```

Comment:
Whenever jvm needs the information of `Observable`, `NoClassDefFoundError` will be thrown. `UnsupportedOperationException` will only be thrown when user inherit the `BaseRealm` and create `Observable` from there. I am not sure why did you throw that at the first place. If you don't recall the reason, maybe we could just remove that?